src/js/components/DataTable/Body.js

```diff
@@ -118,10 +117,17 @@ const Row = memo(
             context={isRowExpanded ? 'groupHeader' : 'body'}
             expanded={isRowExpanded}
             onToggle={() => {
```

Comment:
just wondering if we can make this `onToggle` a little cleaner 


```
           onToggle={() => {
              const rowKey = primaryValue || index;
              const nextRowExpand = isRowExpanded
                ? rowExpand.filter((s) => s !== rowKey)
                : [...rowExpand, rowKey];

              rowDetails.onExpand
                ? rowDetails.onExpand(nextRowExpand, datum)
                : setRowExpand(nextRowExpand);
            }}
            
```

Reply:
How so? It has to conditionally add/remove the key then conditionally call the callback (controlled) or just do the expand change.

Reply:
aw yes conditionally add/remove ignore the comment